app/src/main/java/org/thoughtcrime/securesms/onboarding/recoverypassword/RecoveryPasswordActivity.kt

```diff
@@ -0,0 +1,218 @@
+    }
+}
+
+@Composable
+fun RecoveryPassword(
+    seed: String = "",
+    copySeed:() -> Unit = {},
+    onHide:() -> Unit = {}
+) {
+    AppTheme {
+        Column(
+            verticalArrangement = Arrangement.spacedBy(16.dp),
+            modifier = Modifier
+                .contentDescription(R.string.AccessibilityId_recovery_password)
+                .verticalScroll(rememberScrollState())
+                .padding(bottom = 16.dp)
+        ) {
+            SmallButtons {
+                RecoveryPasswordCell(seed, copySeed)
+                HideRecoveryPasswordCell(onHide)
+            }
+        }
+    }
+}
+
+@Composable
+fun RecoveryPasswordCell(seed: String, copySeed:() -> Unit = {}) {
+    var showQr by remember {
+        mutableStateOf(false)
+    }
+
+    CellWithPaddingAndMargin {
+        Column {
+            Row {
+                Text(stringResource(R.string.sessionRecoveryPassword))
+                Spacer(Modifier.width(8.dp))
+                SessionShieldIcon()
+            }
+
+            Text(stringResource(R.string.recoveryPasswordDescription))
+
+            AnimatedVisibility(!showQr) {
+                Text(
+                    seed,
+                    modifier = Modifier
+                        .contentDescription(R.string.AccessibilityId_recovery_password_container)
+                        .padding(vertical = 24.dp)
+                        .border(
+                            width = 1.dp,
+                            color = classicDarkColors[3],
+                            shape = RoundedCornerShape(11.dp)
+                        )
+                        .padding(24.dp),
+                    textAlign = TextAlign.Center,
+                    style = MaterialTheme.typography.small.copy(fontFamily = FontFamily.Monospace),
+                    color = MaterialTheme.colors.run { if (isLight) onSurface else secondary },
+                )
+            }
+
+            AnimatedVisibility(
+                showQr,
+                modifier = Modifier.align(Alignment.CenterHorizontally)
+            ) {
+                QrImage(
+                    seed,
+                    modifier = Modifier.padding(vertical = 24.dp)
+                        .contentDescription(R.string.AccessibilityId_qr_code),
+                    icon = R.drawable.session_shield
+                )
+            }
+
+            AnimatedVisibility(!showQr) {
+                Row(horizontalArrangement = Arrangement.spacedBy(32.dp)) {
+                    OutlineTemporaryStateButton(
+                        Modifier
+                            .weight(1f)
+                            .contentDescription(R.string.AccessibilityId_copy_button),
+                        onClick = copySeed
+                    ) { isTemporary ->
+                        Text(stringResource( if (isTemporary) R.string.copied else R.string.copy))
```

Comment:
Buttons should have a reusable Text object since they all share the same traits. By adding one here, unstyled, you  are making this button unique and non reusable.

Reply:
Do you need to ever inject other content in here? Or is it always a text. If so you might as well remove the content and pass a string to OutlineTemporaryStateButton, which internally will call the right Outline button ( in this case it is a OutlineButtonSmall )